### PR TITLE
DB-10961 Get rid of the NodeFactory part 2

### PR DESCRIPTION
**Pull request by martinrupp**
iterative PR to get rid of NodeFactory / getNode

### CODE REVIEW

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by martinrupp** [comment]
note this is best reviewed commit by commit. it's just adding constructors and replacing `getNodeFactory().getNode(X, ...)` with `new X(...)` .

**Review by arnaud-splice — CHANGES_REQUESTED** [review]
Great change! One minor request for changes.

**Review by dgomezferro — APPROVED** [review]
Nice simplification! +1 conditioned on the sysout removal

File: db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ResultColumn.java [hunk 1]
```diff
@@ -509,6 +536,9 @@ public int getStoragePosition() {
     public void setExpression(ValueNode expression)
     {
         this.expression = expression;
+        if( expression == null) {
+            System.out.println("ha");
```
[the remark this answers is not in the document]
Comment (martinrupp) [reply db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/ResultColumn.java:540]: thanks done

**Review by arnaud-splice — APPROVED** [review]

**Review by carolp-503 — APPROVED** [review]
+1

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by cloudspliceci** [comment]
TEST SUCCEEDED +1

**Pull request merged**